logging/logrus/payload_interceptors.go

```diff
@@ -31,12 +32,18 @@ func PayloadUnaryServerInterceptor(entry *logrus.Entry, decider grpc_logging.Ser
 			return handler(ctx, req)
 		}
 		// Use the provided logrus.Entry for logging but use the fields from context.
-		logEntry := entry.WithFields(Extract(ctx).Data)
-		logProtoMessageAsJson(logEntry, req, "grpc.request.content", "server request payload logged as grpc.request.content field")
 		resp, err := handler(ctx, req)
-		if err == nil {
-			logProtoMessageAsJson(logEntry, resp, "grpc.response.content", "server response payload logged as grpc.request.content field")
-		}
+
+		fireAndForgetLoggingWaitGroup.Add(1)
+		// Perform payload logging in a goroutien to unblock the response to the client.
+		go func(resp interface{}, err error) {
+			logEntry := entry.WithFields(Extract(ctx).Data)
```

Comment:
Would want to do most of this code outside the go routine as the ctx may change as the app continues.